### ISSUE TITLE
doesn't build on Debian10 microhttpd version 0x00096200 anymore

### ISSUE THREAD

**Issue by ckrey** [jpmens/from-traccar#6]
```
$ make
cc -Wall -Werror -DSTATSDHOST=\""127.0.0.1"\" -DGIT_VERSION=\"0.2-32-gae29f49e72\"   -c -o ft.o ft.c
ft.c:287:18: error: return type is an incomplete type
 static MH_RETVAL on_client_connect(void *cls, const struct sockaddr *addr, socklen_t addrlen)
                  ^~~~~~~~~~~~~~~~~
```

```
$ grep MHD_VERSION /usr/include/microhttpd.h
#define MHD_VERSION 0x00096200
```

**Comment by jpmens** [comment]
that library (libmicrohttpd) is the worst I've ever encountered: I feel as though I'm spending my life patching `ft.c` to adapt to their API changes.

**Comment by jpmens** [comment]
Patched. Could you try again, please?

**Comment by jsponz** [comment]
@jpmens Thanks for the code and support. 

I have just downloaded the code and tried to compile:

```
cc -Wall -Werror -DSTATSDHOST=\""127.0.0.1"\" -DGIT_VERSION=\"0.2-35-gd645d4b60b-dirty\"   -c -o ft.o ft.c
ft.c:286:18: error: return type is an incomplete type
 static MH_RETVAL on_client_connect(void *cls, const struct sockaddr *addr, socklen_t addrlen)
                  ^~~~~~~~~~~~~~~~~
ft.c: In function ‘on_client_connect’:
ft.c:288:9: error: ‘return’ with a value, in function returning void [-Werror]
  return MHD_YES;
         ^~~~~~~
```

I have 

```
grep MHD_VERSION /usr/include/microhttpd.h
#define MHD_VERSION 0x00095900
```

Any help?

**Comment by jpmens** [comment]
I’d appreciate help testing this because I’m not in a position to do so.

Line 15 in `ft.c` currently is

```c
# if MHD_VERSION == 0x00096200 || MHD_VERSION == 0x00093300
```

Can you please change that to read
```c
# if MHD_VERSION == 0x00096200 || MHD_VERSION == 0x00093300 ||  MHD_VERSION == 0x00095900
```

(That last number is the version you reported having.)

**Comment by jsponz** [comment]
It works like a charm!!! Thanks!

```
$ make all
cc -Wall -Werror -DSTATSDHOST=\""127.0.0.1"\" -DGIT_VERSION=\"0.2-35-gd645d4b60b-dirty\"   -c -o ft.o ft.c
cc -Wall -Werror -DSTATSDHOST=\""127.0.0.1"\" -DGIT_VERSION=\"0.2-35-gd645d4b60b-dirty\"   -c -o json.o json.c
cc -Wall -Werror -DSTATSDHOST=\""127.0.0.1"\" -DGIT_VERSION=\"0.2-35-gd645d4b60b-dirty\" -o ft ft.o json.o statsd/statsd-client.o  -lm -L. -lmosquitto -lmicrohttpd
if test -r codesign.sh; then /bin/sh codesign.sh; fi
$ 
```

**Comment by jpmens** [comment]
Thanks for reporting. I will now commit that patch.

**Comment by ckrey** [comment]
A better long term fix would be the same as for jpmens/revgeod#4
[Changelog for libmicrohttpd 0.9.71](https://abi-laboratory.pro/index.php?view=changelog&l=libmicrohttpd&v=0.9.71)
